core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -302,24 +312,19 @@ public String getRunnerSuffix() {
     }
 
     @ConfigGroup
-    public static class VineFlowerConfig {
+    public static class DecompilerConfig {
         /**
          * An advanced option that will decompile generated and transformed bytecode into the 'decompiled' directory.
          * This is only taken into account when fast-jar is used.
          */
-        @ConfigItem(defaultValue = "false")
-        public boolean enabled;
-
-        /**
-         * The version of Vineflower to use
-         */
-        @ConfigItem(defaultValue = "1.9.3")
-        public String version;
+        @ConfigDocDefault("false")
+        public Optional<Boolean> enabled;
```

Comment:
Are these being made into optional booleans solely to accommodate the fact that the configuration moved? If so, I think we can/should probably brainstorm a better solution, otherwise these kinds of things are going to accumulate and cause problems.

Reply:
Correct. I'm definitely open to looking for a better solution

Reply:
A possible solution is to use relocate functions: https://smallrye.io/smallrye-config/3.4.4/extensions/relocate/